target_bigquery.py

```diff
@@ -265,57 +276,276 @@ def persist_lines_stream(project_id, dataset_id, lines=None, validate_records=Tr
+                        )
+                        if (
+                            "payload size exceeds the limit" in error_string
+                            or "too many rows present" in error_string
+                        ):
+                            # Let's try inserting it in two halves
+                            number_of_rows = len(fixed_rows)
+                            errors[stream] = bigquery_client.insert_rows_json(
+                                tables[stream],
+                                fixed_rows[: number_of_rows // 2],
+                                row_ids=ids[: number_of_rows // 2],
+                            )
+                            errors[stream] + bigquery_client.insert_rows_json(
+                                tables[stream],
+                                fixed_rows[number_of_rows // 2 :],
+                                row_ids=ids[number_of_rows // 2 :],
+                            )
+                        elif getattr(e, "errors", [{}])[0].get("reason") in RETRYABLE_ERROR_CODES:
+                            pass
+                        else:
+                            raise e
+
+                    if not errors[stream]:
+                        break
+
+                    sleep(5 if recreated_tables.get(stream) else 1)
+
+                if not errors[stream]:
+                    logger.info(f"Loaded {len(rows[stream])} row(s) into {tables[stream].path}")
+                else:
+                    failed_lines = failed_lines + rows[stream]
+                    logger.warning(
+                        f"Error loading row(s) into '{tables[stream].path}': {str(errors[stream])}",
+                        extra={"stream": stream},
+                    )
+                    del errors[stream]
+
+                recreated_tables.pop(stream, None)
+                rows[stream] = []
+
+        if emit_state_after_write:
+            emit_state(state)
+
+    for line in lines:
+        try:
+            msg = singer.parse_message(line)
+        except json.decoder.JSONDecodeError:
+            logger.warning(f"Unable to parse line: {line}")
+            failed_lines.append(line)
+            continue
+
+        if isinstance(msg, singer.RecordMessage):
+            if msg.stream not in schemas:
+                logger.warning(
+                    f"Record for stream '{msg.stream}' received before its schema!",
+                    extra={"stream": msg.stream},
+                )
+                failed_lines.append(line)
+                continue
+
+            if validate_records:
+                validate(msg.record, schemas[msg.stream])
+
+            rows[msg.stream].append(msg.record)
+
+            state = None
+
+        elif isinstance(msg, singer.StateMessage):
+            state = msg.value
+            full_stream = state.get("currently_syncing") or ""
+            stream = full_stream.split("-")[-1]
```

Comment:
This will error if the above `full_stream = state.get("currently_syncing", "")` gets set to the default value, you'll get an `IndexError` is that expected/wanted?

Reply:
Yeah, so that's why we needed my slightly odd looking `tate.get("currently_syncing") or ""` solution, which I commented in the last commit as I myself forgot why I did it this way.